To: vim_dev@googlegroups.com
Subject: Patch 7.3.096
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.096
Problem:    "gvim -nb" is not interruptable.  Leaking file descriptor on
	    netbeans connection error.
Solution:   Check for CTRL-C typed.  Free file descriptor.  (Xavier de Gaye)
Files:	    src/netbeans.c


*** ../vim-7.3.095/src/netbeans.c	2010-12-24 14:00:09.000000000 +0100
--- src/netbeans.c	2011-01-04 18:00:35.000000000 +0100
***************
*** 321,326 ****
--- 321,327 ----
      {
  	nbdebug(("error in gethostbyname() in netbeans_connect()\n"));
  	PERROR("gethostbyname() in netbeans_connect()");
+ 	sock_close(sd);
  	goto theend;
      }
      memcpy((char *)&server.sin_addr, host->h_addr, host->h_length);
***************
*** 370,384 ****
  							 || (errno == EINTR)))
  		{
  		    nbdebug(("retrying...\n"));
! 		    sleep(5);
! 		    if (!doabort)
  		    {
! 			ui_breakcheck();
! 			if (got_int)
! 			{
! 			    errno = EINTR;
! 			    break;
! 			}
  		    }
  		    if (connect(sd, (struct sockaddr *)&server,
  							 sizeof(server)) == 0)
--- 371,382 ----
  							 || (errno == EINTR)))
  		{
  		    nbdebug(("retrying...\n"));
! 		    mch_delay(3000L, TRUE);
! 		    ui_breakcheck();
! 		    if (got_int)
  		    {
! 			errno = EINTR;
! 			break;
  		    }
  		    if (connect(sd, (struct sockaddr *)&server,
  							 sizeof(server)) == 0)
***************
*** 393,398 ****
--- 391,397 ----
  		    /* Get here when the server can't be found. */
  		    nbdebug(("Cannot connect to Netbeans #2\n"));
  		    PERROR(_("Cannot connect to Netbeans #2"));
+ 		    sock_close(sd);
  		    if (doabort)
  			getout(1);
  		    goto theend;
***************
*** 403,408 ****
--- 402,408 ----
  	{
  	    nbdebug(("Cannot connect to Netbeans\n"));
  	    PERROR(_("Cannot connect to Netbeans"));
+ 	    sock_close(sd);
  	    if (doabort)
  		getout(1);
  	    goto theend;
*** ../vim-7.3.095/src/version.c	2011-01-04 17:49:25.000000000 +0100
--- src/version.c	2011-01-04 18:09:46.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     96,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
85. Choice between paying Compuserve bill and paying for kids education
    is a no brainer -- although a bit painful for your kids.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
